src/bin/jwtdecode.rs

```diff
@@ -32,46 +31,27 @@ fn main() {
         )
         .get_matches();
 
+    let stdin = io::stdin();
     let input_file = matches.value_of("INPUT");
 
-    let results: Vec<String> = match input_file {
+    let input: Box<dyn io::BufRead> = match input_file {
```

Comment:
Fwiw, thought process is:
- these two blocks look similar, I should be able to refactor something here.
- I need a single type for the value at the top because Rust(*).  What's the common thing that each if/else branch "produces".
- The common thing is a trait, not a concrete type so I need an owned trait object (`Box<dyn Trait>`).
- (Bonus) Oh, look.  Once I have a `BufRead`, I don't actually need the rest of the `Vec<String>` loop at all and can just remove it.

(*) Fwiw, another version of this would be to move the rest of `main()` into a subfunction, and call that subfunction from each branch.  The difference is that the subfunction would "borrow" rather than "own" the common thing, so the borrowed-thing could stay on the stack (minor over-optimisation).